lib/plugin-loader.js

```diff
@@ -752,71 +756,275 @@ PluginLoader.prototype = {
     let newPlugins = [];
     let successCount = 0;
     const depgraph = new DependencyGraph(this.plugins);
-    for (const pluginDef of pluginDefs) {
-      if (pluginDef.error) {
-        //downstream will pick up error
-        newPlugins.push(pluginDef);
-      } else {
-        depgraph.addPlugin(pluginDef);
+    const componentCapabilities = await this.getComponentCapabilities(pluginContext.config); // Gets the environment and server information
+
+      for (const pluginDef of pluginDefs) {
+        if (pluginDef.error) {
```

Comment:
This logic (762-772) could be refactored i believe

for ( pluginDef of pluginDefs) {
 this.checkPluginRequirements(pluginDef, componentCapabilities);
 if(pluginDef.error) {
  newPLugins....
 } else {
  deGraph...
 }
}

Reply:
This is a good comment too and intentional. My reasoning here was if there's an error with the plugin before we even get to the installation step, there is no point in going through the steps of checking for plugin requirement problems. Not to mention, if there happened to be a plugin requirement problem, the pluginDef.error would get replaced by this error and the first, more severe error would get overwritten. Let me add a comment for this, this looks very sloppy without the context

if (pluginDef.error) {
          //downstream will pick up error
          newPlugins.push(pluginDef);
        } else {
          this.checkPluginRequirements(pluginDef, componentCapabilities); // Checks that the environment and server satisfies the plugin req's
          if (pluginDef.error) {
            newPlugins.push(pluginDef);
          } else {
            depgraph.addPlugin(pluginDef);
          }
        }

If we find an error un-related to